relax aarch64 relocation checks to allow arm64 linkage with lld

--- a/include/grub/elf.h
+++ b/include/grub/elf.h
@@ -2070,6 +2070,13 @@ typedef Elf32_Addr Elf32_Conflict;
 #define R_AARCH64_ABS64			257	/* Direct 64 bit. */
 #define R_AARCH64_ABS32			258	/* Direct 32 bit.  */
 #define R_AARCH64_PREL32		261
+#define R_AARCH64_MOVW_UABS_G0  263
+#define R_AARCH64_MOVW_UABS_G0_NC 264
+#define R_AARCH64_MOVW_UABS_G1  265
+#define R_AARCH64_MOVW_UABS_G1_NC 266
+#define R_AARCH64_MOVW_UABS_G2  267
+#define R_AARCH64_MOVW_UABS_G2_NC 268
+#define R_AARCH64_MOVW_UABS_G3  269
 #define R_AARCH64_ADR_PREL_PG_HI21	275
 #define R_AARCH64_ADD_ABS_LO12_NC	277
 #define R_AARCH64_LDST64_ABS_LO12_NC	286
--- a/util/grub-module-verifier.c
+++ b/util/grub-module-verifier.c
@@ -111,6 +111,13 @@ struct grub_module_verifier_arch archs[] = {
       R_AARCH64_JUMP26,
       R_AARCH64_ADR_GOT_PAGE,
       R_AARCH64_LD64_GOT_LO12_NC,
+      R_AARCH64_MOVW_UABS_G0,
+      R_AARCH64_MOVW_UABS_G0_NC,
+      R_AARCH64_MOVW_UABS_G1,
+      R_AARCH64_MOVW_UABS_G1_NC,
+      R_AARCH64_MOVW_UABS_G2,
+      R_AARCH64_MOVW_UABS_G2_NC,
+      R_AARCH64_MOVW_UABS_G3,
       -1
     }, (int[]){
       R_AARCH64_ADR_PREL_PG_HI21,
